### PR TITLE
changing search logic from or to and

### PR DESCRIPTION
**Pull request by tejans24**
connects #2231  #2174

### CODE REVIEW

File: client/app/reader/search.js [hunk 1]
```diff
@@ -26,12 +27,28 @@ const tagContainsString = (searchQuery, doc) =>
 
 export const searchString = (searchQuery, state) => (doc) => {
 
-  return !searchQuery || searchQuery.split(' ').some((searchWord) => {
+  let queryTokens = _.compact(searchQuery.split(' '));
+  const dateMatch = doDatesMatch(doc.receivedAt, queryTokens[0]);
```
Comment (tejans24) [inline client/app/reader/search.js]: this assumes date is always the first word in the search query

File: client/app/reader/PdfListView.jsx [hunk 2]
```diff
@@ -11,8 +11,7 @@ import NoSearchResults from './NoSearchResults';
 export class PdfListView extends React.Component {
 
   render() {
-    const noDocuments = !_.size(this.props.documents) &&
-      _.size(this.props.docFilterCriteria.searchQuery);
+    const noDocuments = !_.size(this.props.documents) && _.size(this.props.docFilterCriteria.searchQuery) > 0;
```
Comment (tejans24) [inline client/app/reader/PdfListView.jsx]: This is to fix a console error message in the browser

File: client/test/node/reader/DecisionReviewer-test.js [hunk 3]
```diff
@@ -307,6 +307,17 @@ describe('DecisionReviewer', () => {
     });
 
     context('when searched by', () => {
+      it('does and logic search', () => {
+        wrapper.find('input').simulate('change',
+          { target: { value: '/2017 mytag form' } });
```
Comment (mdbenjam) [inline client/test/node/reader/DecisionReviewer-test.js]: I'd feel more comfortable if we do a second search that has one of these values off to and check if we get 0 results.

File: client/test/node/reader/DecisionReviewer-test.js [hunk 4]
```diff
@@ -410,7 +421,7 @@ describe('DecisionReviewer', () => {
         let textArray = wrapper.find('tr').map((node) => node.text());
 
         // Header and one filtered row.
-        expect(textArray).to.have.length(2);
+        expect(textArray).to.have.length(3);
```
Comment (mdbenjam) [inline client/test/node/reader/DecisionReviewer-test.js]: Change the comments from one filtered row, to two filtered rows.

**Pull request merged**